### PR TITLE
merge perf-dev to main

### PR DESCRIPTION
**Pull request by xiaobaicxy**
## Motivation

1. support graph on npu
2. support mlapo on npu
3. support ep-moe on npu
4. perf opt: dequant-swiglu-quant

## Checklist

- [ ] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

### CODE REVIEW

File: python/sglang/srt/layers/quantization/w8a8_int8.py [hunk 1]
```diff
@@ -572,22 +586,17 @@ def get_perchannel_param(
...
         if original_dtype != torch.int8:
-            x = torch_npu.npu_quantize(
-                x,
-                layer.aclnn_input_scale,
-                layer.aclnn_input_offset,
-                torch.qint8,
-                -1,
-                True,
-            )
+            x = self.elewise_quant.execute([x, layer.input_scale, layer.input_offset])[
```
Comment (shengzhaotian) [inline python/sglang/srt/layers/quantization/w8a8_int8.py:597]: Why replace it with a non-public interface?

File: python/sglang/srt/models/deepseek_v2.py [hunk 2]
```diff
@@ -854,11 +858,11 @@ def __init__(
 
         self.attn_mqa = RadixAttention(
             self.num_local_heads,
-            self.kv_lora_rank + self.qk_rope_head_dim,
```
Comment (xiaobaicxy) [inline python/sglang/srt/models/deepseek_v2.py:857]: 是否影响原有功能

File: python/sglang/srt/models/deepseek_v2.py [hunk 3]
```diff
@@ -986,6 +991,12 @@ def _dispatch_mla_subtype():
         self.current_attention_backend = attention_backend
 
         if attention_backend == "ascend":
+            if (
```
Comment (xiaobaicxy) [inline python/sglang/srt/models/deepseek_v2.py:994]: 参考当前cpu的方法做if分支管理

File: python/sglang/srt/layers/attention/ascend_backend.py [hunk 4]
```diff
@@ -161,62 +234,184 @@ def forward_decode(
...
         save_kv_cache=True,
+        q_rope=None,
     ):
         if save_kv_cache:
             forward_batch.token_to_kv_pool.set_kv_buffer(
                 layer, forward_batch.out_cache_loc, k, v
             )
         if not self.use_mla:
-            k_cache = forward_batch.token_to_kv_pool.get_key_buffer(layer.layer_id)
-            v_cache = forward_batch.token_to_kv_pool.get_value_buffer(layer.layer_id)
```
Comment (RuixuanZhang06) [inline python/sglang/srt/layers/attention/ascend_backend.py:171]: It's unnecessary to wrap the k_cache and v_cache in the if statement.

File: python/sglang/srt/model_executor/npu_graph_runner.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+from __future__ import annotations
+
+import logging
+import threading
+from typing import TYPE_CHECKING
+
+import torch
+
+from python.sglang.srt.model_executor.forward_batch_info import ForwardBatch
```
Comment (iforgetmyname) [inline python/sglang/srt/model_executor/npu_graph_runner.py:24]: wrong import path
